Added Java implementation

Seregy opened the pull request.

Seregy committed:
Added Java implementation

lundibundi commented on Java/SimpleEventEmitter.java:
```diff
@@ -0,0 +1,99 @@
+    private final Map<String, List<Consumer<T>>> events;
+
+    /**
+     * Constructs new SimpleEventEmitter.
```
This comment seems redundant to me.

lundibundi commented on Java/SimpleEventEmitter.java:
```diff
@@ -0,0 +1,99 @@
+    public void emit(final String name, final T data) {
+        List<Consumer<T>> listeners = events.get(name);
+        if (listeners != null) {
+            for (Consumer<T> listener : listeners) {
```
As you use java 8 in this class, I'd suggest you use stream api for this one.

lundibundi commented on Java/SimpleEventEmitter.java:
```diff
@@ -0,0 +1,99 @@
+     *
+     * @param args command-line arguments, not used
+     */
+    public static void main(final String[] args) {
```
I'd like test methods to be in separate class from this lib-class, probably written with junit.

lundibundi reviewed (changes requested):
Overall looks good with a few changes. 

Also I'd suggest to try `Optional<>` with `events` this may simplify other methods but that's not required and I'll LGTM even without it.

tshemsedinov reviewed (changes requested):
Rename files to `N-name.java` like in JS

Seregy committed:
Extracted usage example into separate file, renamed files
Replaced null checks with Optional
Simplified removeAll method
Fixed formatting in comments

Seregy commented:
I've renamed files and made almost all proposed changes. I didn't write test methods because I found it impossible to do with junit inside non-public class(and the class is non-public because of non-conventional file name). So I just moved usage example into separate file. Should I write tests in separate file or pick another testing framework?

lundibundi commented:
@tshemsedinov I'd suggest to rename library class back to something like SimpleEventEmitter as it is done in C# version and the usage/test file name may be in JS style. Because right now it is extremely strange and holds no meaning. Or we can make the library class 'executable' as it was before and name the file 1-simple.

EDIT: @Seregy let's wait for @tshemsedinov to respond to this one and decide based on that.

lundibundi reviewed (approved):
LGTM after resolution of filenames problem. Thanks.
